src/objectives/elbo/scoregradelbo.jl

```diff
@@ -0,0 +1,138 @@
+"""
+	ScoreGradELBO(n_samples; kwargs...)
+
+Evidence lower-bound objective computed with score function gradients. 
+```math
+\\begin{aligned}
+\\nabla_{\\lambda} \\mathrm{ELBO}\\left(\\lambda\\right)
+&\\=
+\\mathbb{E}_{z \\sim q_{\\lambda}}\\left[
+  \\log \\pi\\left(z\\right) \\nabla_{\\lambda} \\log q_{\\lambda}(z)
+\\right]
++ \\mathbb{H}\\left(q_{\\lambda}\\right),
+\\end{aligned}
+```
+
+To reduce the variance of the gradient estimator, we use a baseline computed from a running average of the previous ELBO values and subtract it from the objective.
+
+```math
+\\mathbb{E}_{z \\sim q_{\\lambda}}\\left[
+  \\nabla_{\\lambda} \\log q_{\\lambda}(z) \\left(\\pi\\left(z\\right) - \\beta\\right)
+\\right]
+```
+
+# Arguments
+- `n_samples::Int`: Number of Monte Carlo samples used to estimate the ELBO.
+
+# Keyword Arguments
+- `entropy`: The estimator for the entropy term. (Type `<: AbstractEntropyEstimator`; Default: `ClosedFormEntropy()`)
+- `baseline_window_size::Int`: The window size to use to compute the baseline. (Default: `10`)
+- `baseline_history::Vector{Float64}`: The history of the baseline. (Default: `Float64[]`)
+
+# Requirements
+- The variational approximation ``q_{\\lambda}`` implements `rand` and `logpdf`.
+- `logpdf(q, x)` must be differentiable with respect to `q` by the selected AD backend.
+- The target distribution and the variational approximation have the same support.
+
+Depending on the options, additional requirements on ``q_{\\lambda}`` may apply.
+"""
+struct ScoreGradELBO{EntropyEst <: AbstractEntropyEstimator} <:
+	   AdvancedVI.AbstractVariationalObjective
+	entropy::EntropyEst
+	n_samples::Int
+    baseline_window_size::Int
+    baseline_history::Vector{Float64}
+end
+
+function ScoreGradELBO(
+	n_samples::Int;
+	entropy::AbstractEntropyEstimator = ClosedFormEntropy(),
+    baseline_window_size::Int = 10,
+    baseline_history::Vector{Float64} = Float64[]
+)
+	ScoreGradELBO(entropy, n_samples, baseline_window_size, baseline_history)
+end
+
+
+function Base.show(io::IO, obj::ScoreGradELBO)
+	print(io, "ScoreGradELBO(entropy=")
+	print(io, obj.entropy)
+	print(io, ", n_samples=")
+	print(io, obj.n_samples)
+    print(io, ", baseline_window_size=")
+    print(io, obj.baseline_window_size)
+	print(io, ")")
+end
+
+function compute_control_variate_baseline(history, window_size)
+    if length(history) == 0
+        return 1.0
+    end
+    min_index = max(1, length(history) - window_size)
+    return mean(history[min_index:end])
+end
```

Comment:
I personally think that we should make a whole new set of interface for control variates so that people could easily [mix and match](https://arxiv.org/abs/1810.12482) various control variates. But for now, I think this could be included in the PR.

Reply:
agreed!